### PR TITLE
Wrap each rspec example in a new SDK configuration

### PR DESCRIPTION
**Pull request by flux-ricky**
This caters for examples which explicitly call the tracer provider to shutdown, without doing this subsequent examples have no traces exported.

I hit this issue when using the gem to test a job framework that explicitly tells the tracer provider to shutdown in certain code paths.

This also introduces a flag `:rspec_otel_disable_tracing` which will disable examples from being wrapped in OTEL SDK configuration.

### CODE REVIEW

File: lib/rspec_otel/rspec.rb [hunk 1]
```diff
@@ -3,16 +3,13 @@
...
-
-    OpenTelemetry::SDK.configure do |c|
-      c.add_span_processor span_processor
-    end
-  end
-
   config.around(:each) do |example|
-    example.run
-    RspecOtel.exporter.reset
+    if example.metadata[:rspec_otel_disable_tracing]
```
Comment (dmathieu) [inline lib/rspec_otel/rspec.rb:7]: Could you add documentation of this in the README?

File: spec/rspec_otel/rspec_otel_spec.rb [hunk 2]
```diff
@@ -1,21 +0,0 @@
-# frozen_string_literal: true
```
Comment (flux-ricky) [inline spec/rspec_otel/rspec_otel_spec.rb:1]: I didn't see this file the first time I wrote my tests, I moved these specs into the other RspecOtel spec file which I think is in the more expected location.

File: spec/rspec_otel_spec.rb [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+      described_class.record do
+        expect(described_class.exporter.finished_spans).to be_empty
+        OpenTelemetry.tracer_provider.tracer('rspec-otel').start_span('second').finish
+        expect(described_class.exporter.finished_spans.count).to eq(1)
+        expect(described_class.exporter.finished_spans[0].name).to eq('second')
+      end
+    end
+
+    it "doesn't record unwrapped examples" do
```
Comment (flux-ricky) [inline spec/rspec_otel_spec.rb:42]: Also added some tests down here.

**Comment by flux-ricky** [comment]
Fixed up now @dmathieu

**Comment by flux-ricky** [comment]
If you are happy with this, once merged it'd be great to have another release cut 🙏

File: README.md [hunk 4]
```diff
@@ -73,6 +73,10 @@ Several conditions can be added to the matcher:
 _The `*_event` condition can be called multiple times with different events._
 
 
+## Disabling
```
Comment (dmathieu) [inline README.md:76]:
```suggestion
### Disabling
```

File: README.md [hunk 5]
```diff
@@ -73,6 +73,10 @@ Several conditions can be added to the matcher:
...
+
+RSpec Otel wraps every example in a new OpenTelemetry SDK configuration by default, if you wish to disable this you can tag your example with `:rspec_otel_disable_tracing`.
```
Comment (dmathieu) [inline README.md:78]:
```suggestion
We every example in a new OpenTelemetry SDK configuration by default, if you wish to disable this you can tag your example with `:rspec_otel_disable_tracing`:

```ruby
require 'spec_helper'
RSpec.describe 'User API', :rspec_otel_disable_tracing do
  it 'tests my code' do
    expect(true).to be true
  end
end
```

```

It seems adding a code sample breaks the update suggestion. But I suppose you see the point of adding an example :)

**Review by dmathieu — APPROVED** [review]

**Comment by flux-ricky** [comment]
@dmathieu could I trouble you for a new release of the gem with the latest changes when you have time 🙏 I would like to continue using the upstream project than maintain a fork. Let me know if there is anything I can do to help.

**Pull request merged**